platform/linux-generic/odp_ishm.c

```diff
@@ -238,13 +238,150 @@ typedef struct {
 } ishm_ftable_t;
 static ishm_ftable_t *ishm_ftbl;
 
+#define HP_CACHE_SIZE 32
+struct huge_page_cache {
+	uint64_t len;
+	int total; /* index in fd that's the highest allocated */
+	int idx;   /* retrieve fd[idx] to get a free file descriptor */
+	unsigned int seq_num;
+	int fd[HP_CACHE_SIZE]; /* list of file descriptors */
+};
+
+static struct huge_page_cache hpc;
+
 #ifndef MAP_ANONYMOUS
 #define MAP_ANONYMOUS MAP_ANON
 #endif
 
 /* prototypes: */
 static void procsync(void);
 
+static int hp_create_file(uint64_t len, unsigned int seq_num)
+{
+	char filename[ISHM_FILENAME_MAXLEN];
+	char dir[ISHM_FILENAME_MAXLEN];
+	int fd;
+	void *addr;
+
+	if (len <= 0) {
+		ODP_ERR("Length is wrong\n");
+		return -1;
+	}
+
+	if (!odp_global_data.hugepage_info.default_huge_page_dir) {
+		ODP_ERR("No huge page dir\n");
+		return -1;
+	}
+
+	snprintf(dir, ISHM_FILENAME_MAXLEN, "%s/%s",
+		 odp_global_data.hugepage_info.default_huge_page_dir,
+		 odp_global_data.uid);
+
+	if (mkdir(dir, 0744) != 0) {
+		if (errno != EEXIST) {
+			ODP_ERR("Failed to creatr dir: %s\n", strerror(errno));
+			return -1;
+		}
+	}
+
+	snprintf(filename, ISHM_FILENAME_MAXLEN,
+		 "%s/odp-%d-ishm_cached-%04x",
+		 dir,
+		 odp_global_data.main_pid,
+		 seq_num++);
```

Comment:
Why `seq_num` is increased here again?

Reply:
Correct. It's actually not necessary, this variable is passed by value, so it has no effect whatsoever.